insights/client/connection.py

```diff
@@ -668,44 +695,60 @@ def register(self):
         else:
             return (message, client_hostname, "None", "")
 
-    def upload_archive(self, data_collected, duration):
+    def upload_archive(self, data_collected, content_type, duration):
         """
         Do an HTTPS Upload of the archive
         """
         file_name = os.path.basename(data_collected)
-        try:
-            from insights.contrib import magic
-            m = magic.open(magic.MAGIC_MIME)
-            m.load()
-            mime_type = m.file(data_collected)
-        except ImportError:
-            magic = None
-            logger.debug('python-magic not installed, using backup function...')
-            from .utilities import magic_plan_b
-            mime_type = magic_plan_b(data_collected)
-
-        files = {
-            'file': (file_name, open(data_collected, 'rb'), mime_type)}
-
-        if self.config.analyze_container:
-            logger.debug('Uploading container, image, mountpoint or tarfile.')
-            upload_url = self.upload_url
+        upload_url = self.upload_url
+
+        # platform upload
+        if not self.config.legacy_upload:
+            files = {
+                'upload': (file_name, open(data_collected, 'rb'),
+                           content_type)}
+            headers = {}
+
+        # legacy upload
         else:
-            logger.debug('Uploading a host.')
-            upload_url = self.upload_url + '/' + generate_machine_id()
+            try:
+                from insights.contrib import magic
+                m = magic.open(magic.MAGIC_MIME)
+                m.load()
+                mime_type = m.file(data_collected)
+            except ImportError:
+                magic = None
+                logger.debug(
+                    'python-magic not installed, using backup function...')
+                from .utilities import magic_plan_b
+                mime_type = magic_plan_b(data_collected)
+
+            files = {
+                'file': (file_name, open(data_collected, 'rb'), mime_type)}
+
+            if self.config.analyze_container:
+                logger.debug(
+                    'Uploading container, image, mountpoint or tarfile.')
+            else:
+                logger.debug('Uploading a host.')
+                upload_url = self.upload_url + '/' + generate_machine_id()
+            headers = {'x-rh-collection-time': str(duration)}
 
         logger.debug("Uploading %s to %s", data_collected, upload_url)
 
-        headers = {'x-rh-collection-time': str(duration)}
         net_logger.info("POST %s", upload_url)
         upload = self.session.post(upload_url, files=files, headers=headers)
 
         logger.debug("Upload status: %s %s %s",
                      upload.status_code, upload.reason, upload.text)
         if upload.status_code in (200, 201):
             the_json = json.loads(upload.text)
+        elif upload.status_code == 202:
```

Comment:
I think the answer here is that 200 and 201 represent return codes that include results, but 202 does not include results.  So logging the body of the response seems reasonable because it's gonna be pretty small or empty, but maybe add a bit of context so we know what this log message is.